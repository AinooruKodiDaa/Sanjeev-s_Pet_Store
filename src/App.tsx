import {
  BrowserRouter as Router,
  Routes,
  Route,
  createBrowserRouter,
  RouterProvider,
  useNavigate,
} from "react-router-dom";
import {
  CalendarMonth,
  Dashboard,
  LocalHospital,
  PersonAdd,
  Place,
  TrendingUp,
} from "@mui/icons-material";
import "./App.css";
import AdminDashboardPage from "./pages/AdminDashboardPage";
import ReportPage from "./pages/LandingPage";
import LayoutOne from "./layouts/LayoutOne";
import LoginPage from "./pages/LoginPage";
import {
  AppBar,
  Box,
  Container,
  CssBaseline,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  ThemeProvider,
  Toolbar,
  Typography,
} from "@mui/material";
import { useEffect, useState } from "react";

import {
  AuthProvider,
  NavTabProvider,
  ThemeContextProvider,
  useAuth,
  useThemeContext,
} from "./contexts";
import { useUniqueId } from "./hooks";

import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import SignupPage from "./pages/SignupPage";
import Navbar from "./components/ui/Navbar";
import { ModalProvider } from "./contexts/modalContextProvider";
import { ProtectedRoute } from "./components/specific/ProtectedRoute";
import CustomerDashboardPage from "./pages/CustomerDashboardPage";
import LandingPage from "./pages/LandingPage";

function App() {
  /**theme context */
  const { theme, mode } = useThemeContext();

  /**uuid*/
  const uniqueId = useUniqueId("path-");

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <NavTabProvider>
        <Navbar />

        <Container
          sx={{ paddingBlock: 2, margin: "0 auto", maxWidth: "90rem" }}
        >
          <ToastContainer
            hideProgressBar={true}
            pauseOnHover
            position="bottom-right"
            autoClose={4000}
            theme={mode}
          />
          <Box sx={{ display: { sm: "none" } }}>
            Responsiveness development in progress
          </Box>
          <Router>
            <Routes>
              <Route path="/" element={<LandingPage />} />
              <Route path="/login" element={<LoginPage />} />
              <Route path="/signup" element={<SignupPage />} />
              <Route
                path="/adminDashboard"
                element={
                  <ProtectedRoute>
                    <ModalProvider>
                      <AdminDashboardPage />
                    </ModalProvider>
                  </ProtectedRoute>
                }
              />
              <Route
                path="/customerDashboard"
                element={
                  <ProtectedRoute>
                    <ModalProvider>
                      <CustomerDashboardPage />
                    </ModalProvider>
                  </ProtectedRoute>
                }
              />
            </Routes>
          </Router>
        </Container>
      </NavTabProvider>
    </ThemeProvider>
  );
}

export default App;
