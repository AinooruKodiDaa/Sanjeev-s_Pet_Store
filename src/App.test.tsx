import React from "react";
import { screen } from "@testing-library/react";
import App from "./App";
import { render } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});

test("renders learn react link", () => {
  const wrapper = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
