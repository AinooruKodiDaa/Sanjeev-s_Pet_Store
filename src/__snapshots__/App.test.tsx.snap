// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
LoadedCheerio {
  "0": <header
    class="MuiPaper-root MuiPaper-outlined MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-7alxvo-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-u4fbkj-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-bd8u49-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
      </button>
      <div
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
      >
        React.js Assignment by Sanjeev
      </div>
      <div
        class="MuiBox-root css-1qnu3dg"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorWarning PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorWarning css-izuzoi-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
         
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NotificationsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2m6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1z"
            />
          </svg>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4m0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>,
  "1": <nav
    aria-label="navbar items"
    class="MuiBox-root css-1qy8d9a"
  >
    <div
      class="MuiDrawer-root MuiDrawer-docked css-1qu5pic-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          style="display:flex;flex-direction:column;align-items:center"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <img
              alt="Company Logo"
              height="50"
              src="/images/SGHealthcare.svg"
              width="125"
            />
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1t1c0oo-MuiListItem-root"
            >
              <a
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-30p5bo-MuiButtonBase-root-MuiListItemButton-root"
                href="/"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DashboardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 13h8V3H3zm0 8h8v-6H3zm10 0h8V11h-8zm0-18v6h8V3z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Dashboard
                  </span>
                </div>
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1t1c0oo-MuiListItem-root"
            >
              <a
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-30p5bo-MuiButtonBase-root-MuiListItemButton-root"
                href="/map"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="PlaceIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Track on map
                  </span>
                </div>
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1t1c0oo-MuiListItem-root"
            >
              <a
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-30p5bo-MuiButtonBase-root-MuiListItemButton-root"
                href="/report"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="TrendingUpIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m16 6 2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Report
                  </span>
                </div>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>,
  "2":      ,
  "3": <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  />,
  "4": <div
    class="MuiBox-root css-h3evck"
  >
    <div
      class="MuiBox-root css-14u2lnm"
    >
      <h1>
        Dashboard Page
      </h1>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 5,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
